### ISSUE TITLE
fromOptionProp gives "Argument of type is not assignable to..." error

### ISSUE THREAD

**Issue by ulyssesp** [gcanti/monocle-ts#89]
`fromOptionProp` throws the "is not assignable" TSError. `fromNullableProp` is working as expected. v1.7.1

```
interface A {
    readonly a: Option<number>;
}

const a: Optional<A, number> = Optional.fromOptionProp<A>("a");

interface B {
    readonly b: number | null;
}

const b: Optional<B, number> = Optional.fromNullableProp<B, number, "b">("b");
```

**Comment by gcanti** [comment]
@ulyssesp I can't repro

Code

```ts
import { Option } from 'fp-ts/lib/Option'
import { Optional } from 'monocle-ts'

interface A {
  readonly a: Option<number>
}

const a: Optional<A, number> = Optional.fromOptionProp<A>('a')
```

Versions

TypeScript: 3.3.3333
fp-ts: 1.14.3
monocle-ts: 1.7.1

**Comment by ulyssesp** [comment]
Strange. Manually downgraded to fp-ts@1.14.3 which fixed it, and manually upgrading to fp-ts@1.14.4 didn't break it.
